File: exercise/check.py
import exercise
import json

def test_answer():
    transaction_file = open('./transactions.json')
    transactions = json.load(transaction_file)
    transaction_file.close()
    predicted_amount = exercise.get_predicted_amount(transactions, '2022-04')
    assert isinstance(predicted_amount, int), 'Expected predicted_amount to return integer'
    assert predicted_amount == 9900, 'Expected predicted_amount to return 9900'

File: exercise/check.js
const exercise = require('./exercise');
const transactions = require('./transactions.json');

function main() {
  try {
    const result = exercise.getPredictedAmount(transactions, '2022-04');
    if (!result) {
      return console.log('🟥 Expected `getPredictedAmount` to return something but result is empty');
    }
    if (typeof result !== 'number') {
      return console.log('🟥 Expected `getPredictedAmount` to return a number, got', typeof result)
    }
    if (result !== 9900) {
      return console.log(`🟥 Expected \`getPredictedAmount\` to return 9900, got ${result}`)
    }
    return console.log('🟩 Well done!');
  } catch (error) {
    console.log('🟥 Your code threw an exception:');
    return console.log(error);
  }

}

main();
